### PR TITLE
Update mockImportedComponents to match other projects

### PR DESCRIPTION
**Pull request by lyzadanger**
Also fix a few tests that broke after this change.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/test/LaunchErrorDialog-test.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('LaunchErrorDialog', () => {
     {
       errorState: 'canvas_group_set_not_found',
       expectedText:
-        "Assignment's group set no longer exists in CanvasHypothesis couldn't load this assignment because the assignment's group set no longer exists in Canvas.To fix this problem, an instructor needs to edit the assignment settings and select a new group set.Detailed error info.If the problem persists, send us an email or open a support ticket. You can also visit our help documents.",
+        "Assignment's group set no longer exists in CanvasHypothesis couldn't load this assignment because the assignment's group set no longer exists in Canvas.To fix this problem, an instructor needs to edit the assignment settings and select a new group set.",
```
Comment (lyzadanger) [inline lms/static/scripts/frontend_apps/components/test/LaunchErrorDialog-test.js:68]: Man, these particular tests are fiddly.

Comment (robertknight) [reply lms/static/scripts/frontend_apps/components/test/LaunchErrorDialog-test.js:68]: Having one really long string here is a bit silly. It would probably make more sense for `expectedText` to be an array of strings and then check for the presence of each one.

I assume these changes are required because a component that was not previously being mocked now is with the new version of `mockImportedComponents`?

File: lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js [hunk 2]
```diff
@@ -194,7 +193,7 @@ describe('LMSFilePicker', () => {
       );
       assert.isTrue(tryAgainButton.exists());
 
-      const errorDetails = wrapper.find(ErrorDisplay);
+      const errorDetails = wrapper.find('ErrorDisplay');
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js:196]: Was this change needed because `ErrorDisplay` is now being mocked with the new version of `mockImportedComponents` whereas it wasn't before?

**Review by robertknight — APPROVED** [review]
LGTM. I think it would be a good idea to spin out a `@hypothesis/test-utils` package soon as a shared package for utilities like `mockImportedComponents`, `waitFor` and `waitForElement`.

File: lms/static/scripts/frontend_apps/components/test/LaunchErrorDialog-test.js (hunk 1 above)
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/test/LaunchErrorDialog-test.js:68]: Yes, these tests could use refactor. 

> I assume these changes are required because a component that was not previously being mocked now is with the new version of mockImportedComponents?

That's my assumption. The change to this `mockImportedComponents` included a change in how components are identified (that is, a change beyond the don't-mock-external-components bit), and it seems plausible that a formerly-not-mocked component now _is_ mocked.

File: lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js (hunk 2 above)
Comment (lyzadanger) [reply lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js:196]: Yes, I believe so. In any case, it seemed more sensible to use the component's name string than have a dependency in this test on that component module.
